cfpb/teachers-digital-platform
# Addition functionality for QA day!

Short description explaining the high-level reason for the pull request

## Additions

- Implemented required fields on start page and survey page
- Implemented start over dialog
- Implemented save work dialog
- Enhanced enable / disabled summary button functionality
- Implemented criteria done logic that enabled summary button


## Testing
- Make sure you run ./setup.sh
- Verify the tdp flag is set to true
- *Start Page*
  - Navigate to http://localhost:8000/tdp/crt-start
  - You will see the START page in the screen shot below
  - Fill out the three values and click "Begin Review"
     - You should redirected to the Survey page
     - The below "Debug Values" have been removed for this PR
     - ~~Verify the values are in the "Debug Values" section at the bottom of the page~~
     - ~~NOTE: if you don't see them click the "Start over with a new review" and follow the above steps again.~~
     - If you navigate back to the  start page the values should be pre-populated (as long as you did not first click "Start over with a new review"

- *Survey Page*
  - Navigate to http://localhost:8000/tdp/crt-survey
  - Click each of the buttons at the top and verify the page below it swaps accordingly
    - The Content Distinctive will have different page depending on the value you choose on the start page.  Elementary, Middle, or High school 
      - This is really hard to test because the pages look the same, you have to click some radio buttons, then manually navigate back to the tdp/crt-start page and choose a different grade (with out starting over and clearing db data.  Then verify the content has different values.  The values you choose should persist based on the grade drop down.  (Due to design I had to remove debug values that made this test easy)
  - ~~Notice the Distinctive button menu is now duplicated at the bottom of the screen (It should always mirror the changes that are reflected in the top menu)~~
  - Check some of the radio buttons and fill in some of the text 
  - Verify the selections and text are saved while switching pages
  - Even close the web browser and the selections should be saved
  - Any page you start on the button at the top should state 'in progress'
  - If you click the "Start over with a new review" it should clear all selection and text values and redirect you back to the start page
  - Clicking on "Continue to Summary" will mark the current distinctive as complete and it should have a different styling.
     - To get the "Continue to summary" button enabled you have check every radio button on the page and enter text into both the text fields.


## Review

- @ascott1 @dcmouyard

[Preview this PR without the whitespace changes](?w=0)

## Notes

- Unit Tests are coming... all complex proof's are implemented just need to fill out the rest

## Checklist
